sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -379,7 +379,7 @@ public void setFallbackCoderProvider(CoderProvider coderProvider) {
    *
    * <p>See {@link #setFallbackCoderProvider}.
    */
-  public CoderProvider getFallbackCoderProvider() {
+  private CoderProvider getFallbackCoderProvider() {
```

Comment:
And this one needs to be public so they can combine it with the above. You could replace them both with `addFallbackCoderProvider` so they are only allowed to add alternatives (which should have precedence)